### PR TITLE
chore(wdio): port lifecycle-update test suite to wdio

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This one is pretty straightforward, just one spec.

## What is the current behavior?

GitHub Issue Number: #5458

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

If CI is passing we're good!

### CODE REVIEW

File: test/wdio/lifecycle-update/cmp-a.test.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    expect(loads[6]).toHaveText('lifecycle-update-c componentDidLoad 1');
+    expect(loads[7]).toHaveText('lifecycle-update-b componentDidLoad 1');
+    expect(loads[8]).toHaveText('lifecycle-update-a componentDidUpdate 1');
+    expect(loads[9]).toHaveText('async add child components to lifecycle-update-a 2');
+    expect(loads[10]).toHaveText('lifecycle-update-a componentWillUpdate 2');
+    expect(loads[11]).toHaveText('lifecycle-update-b componentWillLoad 2');
+    expect(loads[12]).toHaveText('lifecycle-update-c componentWillLoad 2');
+    expect(loads[13]).toHaveText('lifecycle-update-c componentDidLoad 2');
+    expect(loads[14]).toHaveText('lifecycle-update-b componentDidLoad 2');
+    expect(loads[15]).toHaveText('lifecycle-update-a componentDidUpdate 2');
```
Comment (christian-bromann) [inline test/wdio/lifecycle-update/cmp-a.test.tsx:61]:
```suggestion
    await expect(loads[0]).toHaveText('lifecycle-update-a componentWillLoad');
    await expect(loads[1]).toHaveText('lifecycle-update-a componentDidLoad');
    await expect(loads[2]).toHaveText('async add child components to lifecycle-update-a 1');
    await expect(loads[3]).toHaveText('lifecycle-update-a componentWillUpdate 1');
    await expect(loads[4]).toHaveText('lifecycle-update-b componentWillLoad 1');
    await expect(loads[5]).toHaveText('lifecycle-update-c componentWillLoad 1');
    await expect(loads[6]).toHaveText('lifecycle-update-c componentDidLoad 1');
    await expect(loads[7]).toHaveText('lifecycle-update-b componentDidLoad 1');
    await expect(loads[8]).toHaveText('lifecycle-update-a componentDidUpdate 1');

    await button.click();

    loads = await $('#output').$$('li');
    await expect(loads).toBeElementsArrayOfSize(16);

    await expect(loads[0]).toHaveText('lifecycle-update-a componentWillLoad');
    await expect(loads[1]).toHaveText('lifecycle-update-a componentDidLoad');
    await expect(loads[2]).toHaveText('async add child components to lifecycle-update-a 1');
    await expect(loads[3]).toHaveText('lifecycle-update-a componentWillUpdate 1');
    await expect(loads[4]).toHaveText('lifecycle-update-b componentWillLoad 1');
    await expect(loads[5]).toHaveText('lifecycle-update-c componentWillLoad 1');
    await expect(loads[6]).toHaveText('lifecycle-update-c componentDidLoad 1');
    await expect(loads[7]).toHaveText('lifecycle-update-b componentDidLoad 1');
    await expect(loads[8]).toHaveText('lifecycle-update-a componentDidUpdate 1');
    await expect(loads[9]).toHaveText('async add child components to lifecycle-update-a 2');
    await expect(loads[10]).toHaveText('lifecycle-update-a componentWillUpdate 2');
    await expect(loads[11]).toHaveText('lifecycle-update-b componentWillLoad 2');
    await expect(loads[12]).toHaveText('lifecycle-update-c componentWillLoad 2');
    await expect(loads[13]).toHaveText('lifecycle-update-c componentDidLoad 2');
    await expect(loads[14]).toHaveText('lifecycle-update-b componentDidLoad 2');
    await expect(loads[15]).toHaveText('lifecycle-update-a componentDidUpdate 2');
```

**Review by christian-bromann — COMMENTED** [review]
😉

File: test/wdio/lifecycle-update/cmp-a.test.tsx (hunk 1 above)
Comment (alicewriteswrongs) [reply test/wdio/lifecycle-update/cmp-a.test.tsx:61]: :O forgot about that whole await thing I guess! 😅

File: test/wdio/lifecycle-update/cmp-a.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          <ol id="output"></ol>
+          <hr />
+          <lifecycle-update-a></lifecycle-update-a>
+        </>
+      ),
+    });
+  });
+
+  it('fire load methods in order', async () => {
+    await $('#output').waitForStable();
```
Comment (rwaskiewicz) [inline test/wdio/lifecycle-update/cmp-a.test.tsx:18]: @christian-bromann Would it make more sense to use `waitForChanges` here from `@wdio/brwoser-runner/stencil`, or does it not matter?

Comment (christian-bromann) [reply test/wdio/lifecycle-update/cmp-a.test.tsx:18]: `waitForChanges` doesn't work and I need to remove it from the package. I initially thought I could port the functionality but it doesn't look like it.

Comment (alicewriteswrongs) [reply test/wdio/lifecycle-update/cmp-a.test.tsx:18]: wondering about this too 🤔

**Review by rwaskiewicz — APPROVED** [review]

**Review by christian-bromann — APPROVED** [review]

**Pull request merged**